test/dynamo/test_backward_higher_order_ops.py

```diff
@@ -121,23 +121,31 @@ def fn(x, y):
                 out.backward(grad_out)
             actual = normalize_gm(graph.print_readable(False))
             self.assertEqual(x.grad, grad_out * grad_out)
-            self.assertExpectedInline(
-                actual,
-                """\
+            if backend == ["aot_eager", "inductor"]:
+                self.assertExpectedInline(
+                    actual,
+                    """\
 class GraphModule(torch.nn.Module):
     def forward(self, L_inputs_ : list):
         l_inputs_ = L_inputs_
 
-        getitem: "f32[s0]" = l_inputs_[0];  l_inputs_ = None
+        getitem: "f32[2]" = l_inputs_[0];  l_inputs_ = None
```

Comment:
this test seems to imply that the post-dynamo graph generated by CA used to have symints but is now specialized. Although I'm actually a bit confused, since there is no usage of dynamic shapes in the testing code. Is it "better" that we are now producing a static shape graph here?

Reply:
I need to dig into this some more

Reply:
So... this is expected from how the validate_outputs call in the compiled autograd graph bakes in a [2] as the shape. I am not sure if that means we lose other dynamic shape opportunities in compiled autograd though